packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -697,12 +697,12 @@ class IrisGridTableModelTemplate<
     return this.dataForCell(x, y)?.format;
   }
 
-  valueForCell(x: ModelIndex, y: ModelIndex): unknown | null {
+  valueForCell(x: ModelIndex, y: ModelIndex): unknown | null | undefined {
     const data = this.dataForCell(x, y);
     if (data) {
       return data.value ?? null;
     }
-    return null;
+    return undefined;
   }
```

Comment:
Should also be returning `undefined` in the `if` case as well.

I think this whole method could just be written like:
```
  valueForCell(x: ModelIndex, y: ModelIndex): unknown | null | undefined {
    return this.dataForCell(x, y)?.value;
  }
```

Reply:
Oh I remember the reason (and please add this as a code comment)

The JS API currently sets null values as undefined in some instances. This means we need to nullish coaelesce so all undefined values from the API return `null` since the data has been fetched. `undefined` is used to indicate the API has not fetched data yet

Reply:
Good catch. I was looking at an older code branch before that fix, my bad.